src-self-hosted/Module.zig

```diff
@@ -175,16 +175,21 @@ pub const Decl = struct {
     /// represents pre- or post- re-analysis.
     generation: u32,
 
-    /// Represents the position of the code in the output file.
-    /// This is populated regardless of semantic analysis and code generation.
-    link: link.File.Elf.TextBlock = link.File.Elf.TextBlock.empty,
-
-    /// Represents the function in the linked output file, if the `Decl` is a function.
-    /// This is stored here and not in `Fn` because `Decl` survives across updates but
-    /// `Fn` does not.
-    /// TODO Look into making `Fn` a longer lived structure and moving this field there
-    /// to save on memory usage.
-    fn_link: link.File.Elf.SrcFn = link.File.Elf.SrcFn.empty,
+    link: union {
+        elf: struct {
+            /// Represents the position of the code in the output file.
+            /// This is populated regardless of semantic analysis and code generation.
+            text_block: link.File.Elf.TextBlock = link.File.Elf.TextBlock.empty,
+
+            /// Represents the function in the linked output file, if the `Decl` is a function.
+            /// This is stored here and not in `Fn` because `Decl` survives across updates but
+            /// `Fn` does not.
+            /// TODO Look into making `Fn` a longer lived structure and moving this field there
+            /// to save on memory usage.
+            src_fn: link.File.Elf.SrcFn = link.File.Elf.SrcFn.empty,
+        },
+        c: void,
+    },
```

Comment:
counter proposal:

```zig
     link: link.File.TextBlock, 
     fn_link: link.File.SrcFn, 
```

 * move the type declaration to link.zig, in `link.File`. (Module.zig is kinda trying to pretend it doesn't care what data is in there)
 * keep them separate - we're closer to solving that TODO with them separate and putting them together in the same struct takes us further away from solving it.